chesster.js

```diff
@@ -51,6 +80,95 @@ var channels = {
         return "<#"+this.getId(name)+">";
     }
 };
+var channelsByID = { };
+
+
+var fuzzy_match = function (search_string, targets) {
+    if (!targets) { return null; } // TODO: not sure this is the right choice ...
+    var results = [];
+    targets.forEach(function(item) {
+        var distance = levenshtein.get(search_string, item);
+        results.push([distance, item]);
+    });
+    results.sort(function(a,b) {
+        return a[0] - b[0];
+    });
+    var choices = [];
+    var min_distance = results[0][0];
+    results.forEach(function(item) {
+        var distance = item[0];
+        var match = item[1];
+        if (distance == min_distance) {
+            choices.push(match);
+        }
+    });
+
+    return choices;
+}
+var get_command_and_targets = function(message, commands, arg_string) {
+    var command = commands[0];
+    var target = "general";
+    var channel = channelsByID[message.channel];
+    if (channel && channel.name) {
+        target = channel.name;
+        console.log("Setting target to: ", channel.name);
+    }
+    var args = arg_string;
+    if (args) {
+        args = args.split(" ");
+        if (args.length > 2) {
```

Comment:
again - I would prefer to throw when I encounter an error. let me know if this is not possible for one reason or another.